### PR TITLE
Optimise support for custom namespaces to use GetMetricData API

### PR DESCRIPTION
**Pull request by code-haven**
Hi! 

This PR optimises the solution in https://github.com/nerdswords/yet-another-cloudwatch-exporter/pull/623 to support ingesting metrics from custom namespaces.  While trying out the original solution, we ran into API throttling issues because it was making too many `GetMetricStatistic` API calls. 

I've adapted the code to use `GetMetricData` instead and  also batch them like how it's done for auto-discovery jobs. 

This fix resulted in the number of API calls to reduce from 100+ `GetMetricStatistic` to just 1 `GetMetricData` API call.

### CODE REVIEW

**Review by datsabk — COMMENTED** [review]
LGTM. Tested locally. Optimises API calls for custom metrics alongside enabling the custom metric based metric data query.

**Review by datsabk — APPROVED** [review]

**Comment by code-haven** [comment]
hey @arielly-parussulo @cristiangreco @thomaspeitz, we're working on deploying YACE to ingest metrics from CloudWatch and stumbled on a requirement to ingest metrics from custom namespaces. 

We noticed that there good work done in this [PR ](https://github.com/nerdswords/yet-another-cloudwatch-exporter/pull/623) to enable this but the solution was making too many AWS API calls causing throttling issues in our account (fixed now).

If the approach in this [commit](https://github.com/nerdswords/yet-another-cloudwatch-exporter/pull/717/commits/743c23fea5d7a92727cdaee4ce4401feb8fe69ad) looks good, do you guys suppose we can use this PR to fast track the release of this feature? I've preserved the original commits and also happy to address pending review comments.

**Comment by cristiangreco** [comment]
@code-haven fine for me if you want to address comments from https://github.com/nerdswords/yet-another-cloudwatch-exporter/pull/623 in this PR, as long as @arielly-parussulo's commits are preserved.

**Comment by code-haven** [comment]
Thank you @cristiangreco. I've address two comments, renamed `customMetrics` usage to `customNamespace`.

File: README.md [hunk 1]
```diff
@@ -410,6 +412,47 @@ static:
...
+| period                 | default value for period                                         |
+| length                 | default value for length                                         |
+| delay                  | default value for delay                                          |
+| addCloudwatchTimestamp | default value for addCloudwatchTimestamp                         |
+
+### Example of config File
+
+```yaml
+apiVersion: v1alpha1
+sts-endpoint: eu-west-1
```
Comment (cristiangreco) [inline README.md:435]:
```suggestion
sts-region: eu-west-1
```

File: pkg/config_test.go [hunk 2]
```diff
@@ -40,6 +41,18 @@ func TestBadConfigs(t *testing.T) {
 			configFile: "unknown_version.bad.yml",
 			errorMsg:   "apiVersion line missing or version is unknown (invalidVersion)",
 		},
+		{
+			configFile: "custom_dimension_without_name.yml",
+			errorMsg:   "Name should not be empty",
```
Comment (cristiangreco) [inline pkg/config_test.go:46]: Nitpick requests:
- test files could be renamed to `custom_namespace_...` now that the config name has changed
- could we keep the pattern of using `.ok.yml` and `.bad.yml` extensions?

Comment (code-haven) [reply pkg/config_test.go:46]: done!

**Review by cristiangreco — APPROVED** [review]

**Pull request merged**